### PR TITLE
Several shell completion fixes

### PR DESCRIPTION
**Pull request by Luap99**
- fix network filters
- add prune filters
- pod create --share support comma separated namespaces

Also fix the volume prune man page which listed invalid prune filters.

### CODE REVIEW

**Comment by Luap99** [comment]
This is the second time this week that dependabot breaks the CI. Both after #10303 and #10316 the Test Code Consistency check was failing due an change in the go.sum file. How can the original PR pass this test but break for all future PRs until the missing line is added to go.sum?

@vrothberg @rhatdan @mheon PTAL

**Comment by TomSweeneyRedHat** [comment]
@Luap99 all kinds of test unhappiness.

**Comment by Luap99** [comment]
@TomSweeneyRedHat The tests are failing because podman.io is down ;(

**Comment by jmguzik** [comment]
@Luap99 @TomSweeneyRedHat 
I opened similar PR a couple of days ago. Could you check how much this ovelaps?
https://github.com/containers/podman/pull/10254

**Comment by Luap99** [comment]
@jmguzik This does not interfere. I merged your PR and rebased this one to also add shell completion for network prune --filter.

**Comment by rhatdan** [comment]
LGTM
@edsantiago PTAL

File: docs/source/markdown/podman-volume-prune.1.md [hunk 1]
```diff
@@ -21,14 +21,9 @@ Do not prompt for confirmation.
 
 #### **--filter**
 
-Filter volumes to be pruned. Volumes can be filtered by the following attributes:
+Filter volumes to be pruned. Volumes can be filtered by the following attribute:
```
Comment (edsantiago) [inline docs/source/markdown/podman-volume-prune.1.md:24]: Is this true? I seem to be able to use every one of those (the removed entries) as an argument to `podman volume prune --filter`.

Comment (Luap99) [reply docs/source/markdown/podman-volume-prune.1.md]: You are right. This was apparently only removed from the compat api endpoint.

**Comment by edsantiago** [comment]
LGTM. If you have to push again (please don't re-push just for this), I think future maintainers might appreciate a checkin comment that includes something like "prune --labels and --until are only stubs; implementing actual completion is TBI".

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests.

**Pull request merged**